torch_xla/csrc/runtime/pjrt_computation_client.cc

```diff
@@ -143,12 +143,16 @@ PjRtComputationClient::PjRtComputationClient() {
         "XLA_COORDINATOR_PORT", XlaCoordinator::kDefaultCoordinatorPort);
 
     // Use the XlaCoordinator as the distributed key-value store.
+    TF_VLOG(3) << "Getting StreamExecutorGpuClient for node_id="
+               << global_process_rank << ", num_nodes=" << global_world_size;
     coordinator_ = std::make_unique<XlaCoordinator>(
         global_process_rank, global_world_size, master_addr, port);
     std::shared_ptr<xla::DistributedRuntimeClient> distributed_client =
         coordinator_->GetClient();
-    auto allowed_devices =
-        std::make_optional<std::set<int>>(std::set{local_process_rank});
+    std::optional<std::set<int>> allowed_devices;
+    if (global_world_size > 1) {
```

Comment:
Synced offline - this is great for single-host single-process development, but in cases where there is a single process per host, this would break in a multihost environment. Outside of SPMD, I'm not aware of a use case for a multihost environment with a single process per host (cc @JackCaoG)

Since we don't officially support SPMD on GPU at the moment, this looks fine to me for now. Once we decide on the right entrypoint for SPMD, we'll need to revisit this.